orca-clouddriver/src/main/java/com/netflix/spinnaker/orca/clouddriver/utils/TrafficGuard.java

```diff
@@ -466,10 +466,9 @@ public boolean hasDisableLock(Moniker clusterMoniker, String account, Location l
     Application application;
     try {
       application = front50Service.get(clusterMoniker.getApp());
-    } catch (RetrofitError e) {
+    } catch (SpinnakerHttpException e) {
```

Comment:
I see a test in TrafficGuardSpec that doesn't strictly need to change because of what's happening in this PR.  It does look like a [broken test](https://github.com/spinnaker/orca/blob/7e923cc05af0f66b952245722aa69c667f685c5c/orca-clouddriver/src/test/groovy/com/netflix/spinnaker/orca/clouddriver/utils/TrafficGuardSpec.groovy#L454) though...and there's a unnecessary null check of application a few lines below here.  Because [Front50Service.get](https://github.com/spinnaker/orca/blob/7e923cc05af0f66b952245722aa69c667f685c5c/orca-front50/src/main/groovy/com/netflix/spinnaker/orca/front50/Front50Service.groovy#L33) returns a "real" type (i.e. not a Response), it never returns null.  So it doesn't make sense to have a test that mocks a null return value.  Can you clean this up please?

Reply:
Looks like a similar thing happens in [UpsertApplicationTaskSpec](https://github.com/spinnaker/orca/blob/7e923cc05af0f66b952245722aa69c667f685c5c/orca-applications/src/test/groovy/com/netflix/spinnaker/orca/applications/tasks/UpsertApplicationTaskSpec.groovy#L60) ...

Reply:
Thanks for [updating TrafficGuardSpec](https://github.com/spinnaker/orca/pull/4623/commits/0c986730d3918caf7b9d8509dea45cb426d9d81a).  The code below still has a null check that doesn't need to be there.

Reply:
> Thanks for [updating TrafficGuardSpec](https://github.com/spinnaker/orca/pull/4623/commits/0c986730d3918caf7b9d8509dea45cb426d9d81a). The code below still has a null check that doesn't need to be there.

Are you talking about this line : https://github.com/Pranav-b-7/orca/blob/0c986730d3918caf7b9d8509dea45cb426d9d81a/orca-clouddriver/src/main/java/com/netflix/spinnaker/orca/clouddriver/utils/TrafficGuard.java#L477 ? 

If yes, this check is still relevant, because in the above catch block when the response code is 404 or 403, application variable is assigned to null.